crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -920,7 +1106,7 @@ async fn mapper_dynamically_updates_supported_operations_for_child_device() {
     let cfg_dir = TempTedgeDir::new();
     create_thin_edge_child_operations(
         &cfg_dir,
-        "child1",
+        "test-device:device:child1",
```

Comment:
Having to fix this test (`mapper_dynamically_updates_supported_operations_for_child_device`) raises questions it would be good to address before merging.

The issue is that the new c8y external ids must be used only over the c8y topics but also for the c8y operation directories. The current behavior might be confusing: the same device being auto-registered differently after an operation file vs on reception of measurement.

Reply:
Even though this directory mechanism won't be around for too long, as long as we support it, it should preserve its former behaviour IMO: where the external user just creates directories with the local device ID, and the mapper should generate the longer external ID with all `<MAIN_DEVICE_ID>:device` prefix. The user shouldn't be forced to create the directories with those prefixes as that is an internal mapper logic.  The child directory name should be treated exactly like the `device_id` in `te/device/<device_id>//` topic.

Reply:
> the external user just creates directories with the local device ID, and the mapper should generate thes  longer external ID with all <MAIN_DEVICE_ID>:device prefix. The user shouldn't be forced to create the directories with those prefixes as that is an internal mapper logic. The child directory name should be treated exactly like the device_id in te/device/<device_id>// topic.

Fully agree. And this needs to be done along this PR.

I think the `children` cache used by the mapper must also be updated to use external-ids for the keys.